docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+
+##### Implementation
+When the user attempts to list the details of a particular category of ingredients, the `listIngredientCommand`, ‘Parser’ and `Ingredient` class will be accessed and the following sequence of actions are called to list details of  a particular category Ingredient list: <br>
+1. User executes `listingredient all` 
+    2. A `Ui` object will be created and calls `Ui#getUserCommand()`
+    3. Input will be parsed in `Command#parseUserCommand()` and identified with the keyword `listingredient`.
+    ![List Ingredient Step 1](images/AddRecipe1.png)
+2. Parsing of user input and creation of command object
+    2.This will automatically trigger the parsing of the user’s input string into a suitable format for the listing of a particular category of  `ingredient` object in `Command#prepareListIngredient()`.
+    3. A `ListIngredientCommand` object will be created.
+    ![List Ingredient Step 2](images/ListIngredientCommand.png)   
+3. Executing Command
+    2. The newly created object will call `#ListIngredientCommand#execute` which starts the process of listing a particular category’s ingredient details, thus calling `ListIngredientCommand#listIngredients()`.
+    3. The existing ingredientList arraylist and the category of the chosen ingredient category  will be passed through to the `ListIngredientCommand#listIngredients()`.
+    4. The function will find if the category name is valid, thus, creates `CommandResult` result storing the details of the ingredient belonging to the particular category.
+    ![List Ingredient Step 3](images/ListIngredientCommand2.png)
+4. The details will then be printed onto the console using `Ui#showResultToUser(result)`.
+    
+
+##### Design Considerations
+Aspect: Finding the category name and print out ingredient belonging to the category
+
+Alternative 1: Looping through the whole ingredientList arraylist to find out all possible category name, then, do sorting and return result
+
+|     |     |
+|-----|-----|
+|**Pros** | The program will be able to detect all category name inside the ingredientList arraylist. |
+|**Cons** | This method will be slow when facing a huge amount of data in the arraylist as the program may have to go through every single item in the arraylist. |
+
+Alternative 2 (current choice): Creating a fixed array which includes the order and all possible category names.
+
+|     |     |
+|-----|-----|
+|**Pros** | Users would be able to get the details of the particular recipe accurately and fast. |
+|**Cons** | Program will not be able to handle any ingredient which isn’t belonging to the category names in the fixed array.  |
+
+#### 4.1.3. Delete all/ specific ingredients(s)
+The deletion feature for ingredients allows the user to delete ingredients either by the name or index of the ingredients. In addition to that, it allows users to reduce the quantity of a specific ingredient. 
+
+![DeleteIngredient Sequence Diagram](images/deleteIngredientSequenceDiagram.png)
```

Comment:
The text is tooooooo small. I suggest split the image and zoom in at the later part when you're explaining about it.